### PR TITLE
Encapsulate parsing algorithm in a helper function, remove writing files

### PR DESCRIPTION
**Pull request by wxli0**
relates to #481 point2, 4
create helper function
remove writing file

### CODE REVIEW

File: src/server/enrichment-map/enrichment/enrichment.js [hunk 1]
```diff
@@ -40,20 +54,13 @@ const enrichment = (query, userSetting) => {
...
-      const str4 = (str1 + '\n').concat(str3);
-      fs.writeFileSync("src/server/enrichment/outputFile", str4);
+
+      fs.writeFileSync("outputFile", parseBody(body));
       // convert csv to json, extract "term id"
       let collection = [];
       let ret = {};
       csv({ delimiter: "\t" })
-        .fromFile("src/server/enrichment/outputFile")
+        .fromFile("outputFile")
```
Comment (d2fong) [inline src/server/enrichment-map/enrichment/enrichment.js]: Not sure if this is what we should be doing

Comment (wxli0) [reply src/server/enrichment-map/enrichment/enrichment.js]: I changed the pr title. will make a separate issue for it next time.

**Comment by d2fong** [comment]
Nah the title is fine this time.  I was talking about the concept of writing files on request from clients that I am not sure if we should be doing.

**Comment by wxli0** [comment]
that might be related to #481 point4. Change the parsing algorithm in the same way as gconvert so that we can avoid creating the csv file.

File: src/server/enrichment-map/enrichment/enrichment.js [hunk 2]
```diff
@@ -13,6 +13,20 @@ const _ = require('lodash');
 const path = require("path");
 
 
+// remove #WARNING and #INFO
+const parseBody = (body) => {
```
Comment (d2fong) [inline src/server/enrichment-map/enrichment/enrichment.js]: use better variable names, e.g:
 parseBody -> parseGProfilerResponse
 body -> gProfilerResponse

These are examples where is is helpful to be less generic

**Pull request merged**